lib/matplotlib/backend_tools.py

```diff
@@ -626,30 +628,56 @@ def scroll_zoom(self, event):
         # https://gist.github.com/tacaswell/3144287
         if event.inaxes is None:
             return
-        ax = event.inaxes
-        cur_xlim = ax.get_xlim()
-        cur_ylim = ax.get_ylim()
-        # set the range
-        cur_xrange = (cur_xlim[1] - cur_xlim[0])*.5
-        cur_yrange = (cur_ylim[1] - cur_ylim[0])*.5
-        xdata = event.xdata  # get event x location
-        ydata = event.ydata  # get event y location
+
+        scl = self.base_scale
         if event.button == 'up':
             # deal with zoom in
-            scale_factor = 1 / self.base_scale
+            direction = 'in'
         elif event.button == 'down':
             # deal with zoom out
-            scale_factor = self.base_scale
+            direction = 'out'
         else:
             # deal with something that should never happen
-            scale_factor = 1
-        # set new limits
-        ax.set_xlim([xdata - cur_xrange*scale_factor,
-                     xdata + cur_xrange*scale_factor])
-        ax.set_ylim([ydata - cur_yrange*scale_factor,
-                     ydata + cur_yrange*scale_factor])
+            scl = 1
+
+        ax = event.inaxes
+        # note: This code needs to be eventually replaced by axes independent
+        # code to obtain the bounding box coordinates.
+        xmin, xmax = ax.get_xlim()
```

Comment:
I am still using get_xlim() and get_ylim() which depends on an orthogonal set of axes here. The main reason is that there is no way to obtain the bounding box.
If I want an axes independent zoom, I either:
1. need to obtain the bounding box coordinates from a function that retrieves it (in the axes perhaps)
2. need to call a function that will do the zooming for me (again probably put in axes)

I believe that without any one of these two possibilities (which I do not see as of now), it is not possible to make a tool that zooms and is independent of the choice of axes.
My personal suggestion is to enforce that the axes coordinates be normalized bounding box coordinates. I don't see why axes coordinates would otherwise be useful for since there is already data coordinates.

There are currently three coordinate transforomations: canvas, axes and data. This would mean that canvas *and* axes would be some universal coordinate system that new coordinate systems should conform to while the data coordinates could be left arbitrary (since we would not longer need "set_xlim" and "set_ylim").

So I am asking, is there discussion on the transformations? Given there is more development on coordinate systems such as polar coordinates, I think it's worth revisitng and defining them more rigorously. This is the main reason why I am stuck on using "get_xlim()" thanks.